block/manager.go

```diff
@@ -249,6 +243,56 @@ func (m *Manager) AggregationLoop(ctx context.Context, lazy bool) {
 	}
 }
 
+func (m *Manager) SetFraudProofService(fraudProofServ *fraudserv.ProofService) {
+	m.executor.SetFraudProofService(fraudProofServ)
+}
+
+func (m *Manager) ProcessFraudProof(fraudProofService *fraudserv.ProofService, ctx context.Context, cancel context.CancelFunc) {
```

Comment:
why does the `fraudProofService` need to be an input? we have `SetFraudProofService` so the `Manager` should already have access to the `fraudProofService`.

Wouldn't it be better to have a check in this function that a fps was set in the first place?

Reply:
does the `executor` also need the fps (`fraudProofService`) for `VerifyFraudProof`?

fi so, then even though this is the only place the manager needs the fps, we could be passing in a different fps than what the executor is using in `VerifyFraudProof`.